packages/strapi-plugin-users-permissions/controllers/Auth.js

```diff
@@ -206,7 +206,12 @@ module.exports = {
       });
 
       // Update the user.
-      await strapi.query('user', 'users-permissions').update({ id: user.id }, user);
+      await strapi
+        .query('user', 'users-permissions')
+        .update(
+          { id: user.id },
+          { resetPasswordToken: user.resetPasswordToken, password: user.password }
```

Comment:
You can avoid mutating the user her. and just used two variables for reset token and password